sys/net/gnrc/netif/gnrc_netif_ieee802154.c

```diff
@@ -94,6 +94,10 @@ static gnrc_pktsnip_t *_recv(gnrc_netif_t *netif)
         if (nread <= 0) {
             gnrc_pktbuf_release(pkt);
             return NULL;
+        } else if (nread <= 1) {
+            DEBUG("_recv_ieee802154: received frame is too short\n");
+            gnrc_pktbuf_release(pkt);
+            return NULL;
```

Comment:
True, but this can be merged with the if-statement above, since `nread <= 1` ⇒ `nread <= 0`

Reply:
Mhhh thinking about it, isn't this already too late for that? I would rather abort already, when `bytes_expected` is `<= 1` + I need to read up on IEEE 802.15.4 if there might be legal frames that are 1 byte long...

Reply:
> True, but this can be merged with the if-statement above, since nread <= 1 ⇒ nread <= 0

Sure, but I thought that it would be a good idea to emit some debug output when discarding a frame due to it being to short.

> I would rather abort already, when bytes_expected is <= 1

I also considered doing that, however, if I understood `dev->driver->recv` correctly the frame is not removed from the buffer when invoking `dev->driver->recv(dev, NULL, 0, NULL)` if that's not the case we could also change the if statement in line 85.

Reply:
> Adding an extra branch seemed cleaner than changing the original if-statement to:
> ```C
> /* … */
> if (nread <= 1) {
>   if (nread == 1) DEBUG("_recv_ieee802154: received frame is too short\n");
>   gnrc_pktbuf_release(pkt);
>  return NULL;
> }
> /* … */
> ```

Debugging is optimized out when disable, so the extra branch potentially causes a larger code size. To help with optimization you could surround the `if (nread == 1) ...` with an `#if ENABLE_DEBUG` (and please keep in line with coding conventions ⇒ no oneline `if`)

> I also considered doing that, however, if I understood dev->driver->recv correctly the frame is not removed from the buffer when invoking dev->driver->recv(dev, NULL, 0, NULL) if that's not the case we could also change the if statement in line 85.

Yes, but you can remove it with `dev->driver->recv(dev, NULL, num, NULL)` with `num > 0`.

Reply:
>  I need to read up on IEEE 802.15.4 if there might be legal frames that are 1 byte long...

According to IEEE802.15.4-2015 the FCF *can* be 1 byte for Multipurpose and Extended frames, but we currently do not support these, so I think this check makes sense for now. However, since the sequence number at the moment is also always to be included, maybe it makes sense to check even if `nread` (or `bytes_expected` if we go for that) is `< 3` (2 byte FCF + 1 byte sequence number).

Reply:
> Yes, but you can remove it with dev->driver->recv(dev, NULL, num, NULL) with num > 0.

Good idea, updated the diff accordingly. Let me know what you thing and I will squash this.

Reply:
> [...] maybe it makes sense to check even if nread (or bytes_expected if we go for that) is < 3 (2 byte FCF + 1 byte sequence number).

Somehow I didn't read this comment before I wrote my previous reply, I can change the value of the `IEEE802154_MIN_FRAME_LEN` macro accordingly :)

Reply:
Yes, please

Reply:
Done.